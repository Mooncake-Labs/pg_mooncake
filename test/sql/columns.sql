CREATE TABLE t (a int DEFAULT NULL, b int) USING columnstore;
INSERT INTO t VALUES (DEFAULT, 1), (DEFAULT, 2);
INSERT INTO t VALUES (3, 3), (4, 4);
INSERT INTO t SELECT 5, 5;
SELECT * FROM t;
DROP TABLE t;

CREATE TABLE t (a int DEFAULT 3, b int) USING columnstore;
INSERT INTO t VALUES (DEFAULT, 1), (DEFAULT, 2);
INSERT INTO t VALUES (3, 3), (4, 4);
INSERT INTO t SELECT 5, 5;
SELECT * FROM t;
DROP TABLE t;

CREATE TABLE t (a int DEFAULT 1 + 2, b int) USING columnstore;
INSERT INTO t (b) VALUES (1), (2);
DROP TABLE t;

CREATE TABLE t (a int GENERATED ALWAYS AS IDENTITY, b int) USING columnstore;
INSERT INTO t (b) VALUES (1), (2);
INSERT INTO t (b) SELECT 3;
SELECT * FROM t;
DROP TABLE t;

CREATE TABLE t (a int) USING columnstore;
INSERT INTO t VALUES (1), (2);
CREATE TABLE r (b int DEFAULT 1 + 2);
INSERT INTO r VALUES (1), (2);
SELECT * FROM t JOIN r ON a = b;
DROP TABLE t, r;
