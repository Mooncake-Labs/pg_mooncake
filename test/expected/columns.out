CREATE TABLE t (a int DEFAULT NULL, b int) USING columnstore;
INSERT INTO t VALUES (DEFAULT, 1), (DEFAULT, 2);
INSERT INTO t VALUES (3, 3), (4, 4);
INSERT INTO t SELECT 5, 5;
SELECT * FROM t;
 a | b 
---+---
   | 1
   | 2
 3 | 3
 4 | 4
 5 | 5
(5 rows)

DROP TABLE t;
CREATE TABLE t (a int DEFAULT 3, b int) USING columnstore;
INSERT INTO t VALUES (DEFAULT, 1), (DEFAULT, 2);
INSERT INTO t VALUES (3, 3), (4, 4);
INSERT INTO t SELECT 5, 5;
SELECT * FROM t;
 a | b 
---+---
 3 | 1
 3 | 2
 3 | 3
 4 | 4
 5 | 5
(5 rows)

DROP TABLE t;
CREATE TABLE t (a int DEFAULT 1 + 2, b int) USING columnstore;
INSERT INTO t (b) VALUES (1), (2);
ERROR:  (PGDuckDB/CreatePlan) Prepared query returned an error: 'Not implemented Error: column "a" has unsupported default value 
DROP TABLE t;
CREATE TABLE t (a int GENERATED ALWAYS AS IDENTITY, b int) USING columnstore;
INSERT INTO t (b) VALUES (1), (2);
INSERT INTO t (b) SELECT 3;
SELECT * FROM t;
 a | b 
---+---
 1 | 1
 2 | 2
 3 | 3
(3 rows)

DROP TABLE t;
CREATE TABLE t (a int) USING columnstore;
INSERT INTO t VALUES (1), (2);
CREATE TABLE r (b int DEFAULT 1 + 2);
INSERT INTO r VALUES (1), (2);
SELECT * FROM t JOIN r ON a = b;
 a | b 
---+---
 1 | 1
 2 | 2
(2 rows)

DROP TABLE t, r;
